app/src/App.tsx

```diff
@@ -23,132 +22,144 @@ import { Guest } from './models'
 import { AppConfig } from './data/config'
 import { useAuth0 } from './react-auth0-spa'
 import {
-    CreateProfile,
-    CreateHostProfile,
-    CreateGuestProfile,
+  CreateProfile,
+  CreateHostProfile,
+  CreateGuestProfile,
 } from './pages/CreateProfile'
 import { AllHosts } from './pages/Admin/AllHosts'
 import ProfileSelection from './pages/ProfileSelection/ProfileSelection'
-import HostQuestionsPage from './pages/HostQuestions'
-import GuestQuestionsPage from './pages/GuestQuestions'
 
-export interface AppProps {}
+import HostQuestions from './pages/HostQuestions'
+import { Fetcher } from './data/ApiWrapper'
+import { Host } from './models/Host'
+
+export interface AppProps { }
 
 export const LoginView = () => {
-    const { loginWithPopup } = useAuth0()
+  const { loginWithPopup } = useAuth0()
 
-    const onLoginClick: React.EventHandler<React.SyntheticEvent<
-        HTMLAnchorElement
-    >> = async (e: React.SyntheticEvent<HTMLAnchorElement>) => {
-        e.preventDefault()
+  const onLoginClick: React.EventHandler<React.SyntheticEvent<
+    HTMLAnchorElement
+  >> = async (e: React.SyntheticEvent<HTMLAnchorElement>) => {
+    e.preventDefault()
 
-        await loginWithPopup()
-    }
+    await loginWithPopup()
+  }
 
-    return (
-        <AppStyle.AuthHolder>
-            <AppStyle.AuthButton href="" onClick={onLoginClick}>
-                Login to Host Homes
-            </AppStyle.AuthButton>
-        </AppStyle.AuthHolder>
-    )
+  return (
+    <AppStyle.AuthHolder>
+      <AppStyle.AuthButton href="" onClick={onLoginClick}>
+        Login to Host Homes
+      </AppStyle.AuthButton>
+    </AppStyle.AuthHolder>
+  )
 }
 
 export const App = () => {
-  // const history = useHistory()
+  const history = useHistory()
+  const { isInitializing, isAuthenticated, user, logout } = useAuth0();
+  const [hasAccount, setHasAccount] = React.useState(false)
 
   const logoutClick = () => {
-    logout()
-    // history.push('/') //route not working but you get signed out
+    logout() //something has to be updated on auth0 side to go to the right route
+
   }
 
-  const { isInitializing, isAuthenticated, user, logout } = useAuth0();
+  React.useEffect(() => {
+    const fetch = async () => {
+      if (isAuthenticated) {
+        let fetch = new Fetcher<Host>('checkEmail')
+        let data: any | undefined = await fetch.getByEmail(user)
```

Comment:
This can be made more concise in a future refactor. Since we already used the email to query the back end, and the back end determined whether this email had a registered account, we can just use the appropriate HTTP status code to indicate whether the account was found or not. A response payload only needs to be read if an existing account was found (HTTP 20*), which provides the account ID and type for subsequent routing to user's dashboard

Reply:
you're right about using status code rather than using the email itself